### PR TITLE
Switch to using file locations in app-config.yaml

### PR DESCRIPTION
**Pull request by Fox32**
Having file locations makes the local workflow of editing entities, e.g. for testing much easier. When you add new files, you don't run into a circular dependency anymore, previously you couldn't test whether the configuration worked before merging it into master.

It also resolves the hard dependency on the GitHub repository - which allows to do "breaking changes" to the files without having to worry that existing instances used for development locally will break. However after this change we should have a grace period in that we should still be careful.

But I didn't performed this change for `create-app`, because it doesn't make sense for new standalone apps. Therefore the previous paragraph is not entirely true.

Beside that there might be hidden risks that might break others workflows. We have this configuration locally for a long time and it works well.

No changeset I guess? 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
I agree that this is likely better, since there's been some confusion around edited files not showing being reflected too.

It potentially breaks some deployments flows from this repo but tbh that shouldn't matter much. A reason for not doing this earlier is because the file paths are pretty weird and we didn't really intend to support relative paths 😅 

One thing to explore would be how we could make them relative to the config file location rather than the working dir, but that can come later.

Happy with trying this out as it's also quite simple to revert and only has an impact on this repo

File: app-config.yaml [hunk 1]
```diff
@@ -185,33 +185,35 @@ catalog:
...
     # Backstage example templates
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/all-templates.yaml
+    - type: file
+      target: ../../plugins/scaffolder-backend/sample-templates/all-templates.yaml
     # Backstage example groups and users
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
+    - type: file
+      target: ../catalog-model/examples/acme-corp.yaml
```
Comment (OrkoHunter) [inline app-config.yaml:216]: I think we should add at least one `- type: url` (commented out is fine) to show an example if someone wants to quickly add a new location with type `url`. :)

Comment (Fox32) [reply app-config.yaml:227]: 👍  For sure, added some more comments

**Comment by Fox32** [comment]
> One thing to explore would be how we could make them relative to the config file location rather than the working dir, but that can come later.

@Rugvip how would resolve the config file location? What if I specify `--config` more than once?

File: app-config.yaml (hunk 1 above)
Comment (OrkoHunter) [reply app-config.yaml:227]: Thank you!

**Comment by Rugvip** [comment]
@Fox32 additional metadata within `ConfigReader` most likely, would be relative to each file. Could also take the form of a secret-like thing that is resolved when loading, like `$path`. Awkward af anyway, let's not try to address that now and merge this :grin:

**Pull request merged**